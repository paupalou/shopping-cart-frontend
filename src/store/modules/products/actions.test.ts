import {
  fetchAllProductsRequestAction,
  fetchAllProductsSuccessAction,
  fetchAllProductsFailureAction,
  incrementProductQuantityAction,
  decrementProductQuantityAction
} from './actions';

describe('Products Actions', () => {
  it('Request action', () => {
    expect(fetchAllProductsRequestAction()).toMatchSnapshot();
  });

  it('Request Success action', () => {
    const products = [
      { id: 1, name: 'Tomato', customerPrice: 50 },
      { id: 2, name: 'Banana', customerPrice: 90 }
    ];

    expect(fetchAllProductsSuccessAction(products)).toMatchSnapshot();
  });

  it('Request Failure action', () => {
    expect(fetchAllProductsFailureAction()).toMatchSnapshot();
  });

  it('Increment quantity action', () => {
    const itemId = 1;
    expect(incrementProductQuantityAction(itemId)).toMatchSnapshot();
  });

  it('Decrement quantity action', () => {
    const itemId = 1;
    expect(decrementProductQuantityAction(itemId)).toMatchSnapshot();
  });
});
