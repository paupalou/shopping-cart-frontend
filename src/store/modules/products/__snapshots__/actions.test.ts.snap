// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Actions Decrement quantity action 1`] = `
Object {
  "id": 1,
  "type": "DECREMENT_PRODUCT_QUANTITY",
}
`;

exports[`Products Actions Increment quantity action 1`] = `
Object {
  "id": 1,
  "type": "INCREMENT_PRODUCT_QUANTITY",
}
`;

exports[`Products Actions Request Failure action 1`] = `
Object {
  "type": "FETCH_ALL_PRODUCTS_FAILURE",
}
`;

exports[`Products Actions Request Success action 1`] = `
Object {
  "payload": Array [
    Object {
      "customerPrice": 50,
      "id": 1,
      "name": "Tomato",
    },
    Object {
      "customerPrice": 90,
      "id": 2,
      "name": "Banana",
    },
  ],
  "type": "FETCH_ALL_PRODUCTS_SUCCESS",
}
`;

exports[`Products Actions Request action 1`] = `
Object {
  "type": "FETCH_ALL_PRODUCTS_REQUEST",
}
`;
